src/wp-includes/class-wp-theme.php

```diff
@@ -361,10 +361,10 @@ public function __construct( $theme_dir, $theme_root, $_child = null ) {
 		if ( ! $this->template ) {
 			$this->template = $this->stylesheet;
 			$theme_path     = $this->theme_root . '/' . $this->stylesheet;
+			$this->is_block_theme();
 
 			if (
-				! file_exists( $theme_path . '/templates/index.html' )
-				&& ! file_exists( $theme_path . '/block-templates/index.html' ) // Deprecated path support since 5.9.0.
+				! $this->block_theme
```

Comment:
@spacedmonkey Actually, can't we just make this `! $this->is_block_theme()`, then since we already know it's `false`, just explicitly pass `false` in the array below?

Reply:
If we call it in the conditional, then we could not need to call the method in the line above. The other reference can be left as a reference to the class property

Reply:
Yep!